integration-tests/slack/notify-circleci-test-results.js

```diff
@@ -0,0 +1,71 @@
+const fs = require('fs')
+const _ = require('lodash/fp')
+const { getFailedTestNamesFromArtifacts } = require('./circleci-utils')
+const { getMessageBlockTemplate } = require('./message-templates')
+const { postMessage } = require('./post-message')
+
+
+/**
+ * Get a map object of Slack channel IDs and array of test names for failed tests notification
+ * @param { Array[string] } failedTests
+ * @returns { Map<string, Array[string]> } A map object where key is channel_id, value is test_names array
+ */
+const getFailedTestsAndChannelIDs = failedTests => {
+  const { fail: jsonBlock } = JSON.parse(fs.readFileSync('./slack/slack-notify-channels.json', 'utf8'))
+  const idsAndNames = new Map() // Map<string, Array[string]>
+  _.forEach(item => {
+    const channelId = _.get('id', item)
+    const { tests } = item
+    const testNames = _.map(test => test.name, tests)
+    idsAndNames.set(channelId, testNames)
+  }, jsonBlock)
+
+  const filteredIDsAndNames = new Map() // Map<string, Array[string]>
+  _.forEach(test => {
+    const channelIdsForFailedTest = Array.from(idsAndNames.keys())
+      .filter(key => idsAndNames.get(key).includes(test) || idsAndNames.get(key).length === 0)
+    if (channelIdsForFailedTest.length === 0) {
+      throw new Error(`Test: ${test} was not found in slack-notify-channels.json`)
+    }
+    _.forEach(channelId => filteredIDsAndNames.has(channelId) ? filteredIDsAndNames.get(channelId).push(test) : filteredIDsAndNames.set(channelId, new Array(test)),
+      channelIdsForFailedTest
+    )
```

Comment:
This is opinion, but I think the arrow function and ternary suggest that the return value here is important, when really we care about the side effects (updating the map and array with set and push). I'll claim this would be clearer as something like:

```suggestion
    _.forEach(channelId => {
      if (!filteredIDsAndNames.has(channelId)) {
        filteredIDsAndNames.set(channelId, [])
      }
      filteredIDsAndNames.get(channelId).push(test)
    }, channelIdsForFailedTest)
```